### ISSUE TITLE
SC1111 - Don’t warn when quotes occur within string literal

### ISSUE THREAD

**Issue by twhb** [koalaman/shellcheck#3464]
#### For new checks and feature suggestions

- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows a potential problem:

``` sh
#!/bin/sh
printf "error: “$var_name” has value “$var_value”\n"
```

#### Here's what shellcheck currently says:

> [SC1111](https://www.shellcheck.net/wiki/SC1111) (warning): This is a unicode quote. […]

#### Here's what I wanted to see:

I’d like SC1111 to be able to recognize that this specific usage of Unicode quotes is valid and probably intentional, and produce no warning. Rewriting the code into something that SC1111 is happy with actually makes it worse, I think:

``` sh
printf 'error: “'"$var_name"'” has value “'"$var_value"'”\n'
```

**Comment by brother** [comment]
Perfect case for a ignore directive.

**Comment by twhb** [comment]
> Perfect case for a ignore directive.

This is indeed what I’m doing now, I just thought it might be appreciated to suggest a way to make it a little bit better for everybody.
